src/Diagnostics.ModelsAndUtils/ScriptUtilities/Utilities.cs

```diff
@@ -121,5 +122,64 @@ public static string HostNamesFilterQuery(IEnumerable<string> hostnames, string
+        /// Filter the given table to get site events for the given slot name.
+        /// </summary>
+        /// <param name="slotName">SlotName</param>
+        /// <param name="slotTimeRanges">Runtime site slot map obtained from observer</param>
+        /// <param name="siteEventsTable">DataTable of web app events that atleast has a site name column and a timestamp column</param>
+        /// <param name="siteColumnName">Name of column for site name</param>
+        /// <param name="timeStampColumnName">Name of column for timestamp</param>
+        /// <example>
+        /// <code>
+        /// public async static Task<Response> Run(DataProviders dp, <![CDATA[OperationContext<App> cxt]]>, Response res){
+        ///     var siteEventsDataTable = await dp.Kusto.ExecuteQuery(runtimeWorkerEventsQuery, cxt.Resource.Stamp.InternalName));
+        ///     var slotTimeRanges = await dp.Observer.GetRuntimeSiteSlotMap(cxt.Resource.Stamp.InternalName, cxt.Resource.Name);
+        ///     var webAppStagingEvents = Utilities.GetSlotEvents(cxt.Resource.Slot, slotTimeRanges, siteEventsDataTable, "SiteName", "TIMESTAMP").
+        /// }
+        /// </code>
+        /// </example>
+        /// <returns></returns>
+        public static DataTable GetSlotEvents(string slotName, Dictionary<string, List<RuntimeSitenameTimeRange>> slotTimeRanges, DataTable siteEventsTable, string siteColumnName = "SiteName", string timeStampColumnName = "TIMESTAMP")
+        {
+            var dt = new DataTable();
+            var columns = siteEventsTable.Columns.Cast<DataColumn>().Select(dc => new DataColumn(dc.ColumnName, dc.DataType, dc.Expression, dc.ColumnMapping)).ToArray();
+            List<RuntimeSitenameTimeRange> timeRangeForSlot;
+
+            if (slotTimeRanges.ContainsKey(slotName.ToLower()))
+            {
+                timeRangeForSlot = slotTimeRanges[slotName];
+            }
+            else
+            {
+                throw new ArgumentException($"{slotName} is not an existing slot for this site. ");
+            }
+
+            try
+            {
+                dt.Columns.AddRange(columns);
+
+                foreach (DataRow row in siteEventsTable.Rows)
+                {
+                    var siteName = (string)row[siteColumnName];
+                    var timeStamp = (DateTime)row[timeStampColumnName];
+
+                    foreach (var slotTimeRangeInfo in timeRangeForSlot)
+                    {
+                        if (siteName.Equals(slotTimeRangeInfo.RuntimeSitename, StringComparison.CurrentCultureIgnoreCase) && timeStamp >= slotTimeRangeInfo.StartTime && timeStamp <= slotTimeRangeInfo.EndTime)
+                        {
+                            dt.Rows.Add(row.ItemArray);
+                            break;
+                        }
+                    }
+                }
+            }catch(Exception ex)
+            {
+                throw new Exception("Failed to get runtime site events table.", ex);
```

Comment:
maybe include 'runtime slotmap' in the message as now that is a well-known term in our team and we can identify quickly which method is throwing this exception.